content/articles/oop-in-java-abstraction-and-encapsulation/index.md

```diff
@@ -1,350 +1,482 @@
+
 A parent class is inherited by a child class known as the client code which overrides the parent’s method signature and provides a method body.
 
 ```java
+
 public interface Car{
-	public void start();
 
-    public void stop();
+   public void start();
+
+   public void stop();
+
 } 
 
 public Benz implements Car{
-    
-    /*@Override to override the stop method in the car class that has no implementaion*/
 
-	@Override 
-	public void start(){
-	    System.out.println(“I started my Benz car ”);
-	}
+   /*@Override to override the stop method in the car class that has no implementation*/
 
-    /*to override the stop method in the car class that has no implementaion*/
+   @Override 
+
+   public void start(){
+
+       System.out.println(“I started my Benz car ”);
+
+   }
+
+   /*to override the stop method in the car class that has no implementation*/
+
+   @Override 
+
+   public void stop(){
+
+       System.out.println(“I stopped my Benz car”)
+
+   }
 
-    @Override 
-	public void stop(){
-		System.out.println(“I stopped my Benz car”)
-    }
 }
+
 ```
 
-To run this program, we need another class and the main method,the latter is the entrance to every Java application.
+To run this program, we need another class and the main method, the latter is the entrance to every Java application.
 
 ```java
+
 public class Main{
-	public static void main(){
-		Car car = new Benz();
-		car.start();
-		car.stop();
-    }
+
+   public static void main(){
+
+       Car car = new Benz();
+
+       car.start();
+
+       car.stop();
+
+   }
+
 }
+
 ```
 
-In the main method, Car represents the interface class or superclass, Benz represents the concrete class or child class, the child class gives its memory to the Car object to have access to its implemented methods. The Benz class implemented the start and stop behaviors of the Car class in the manner it wanted. This makes for application robustness, we can easily change or add a requirement to our application without necessarily breaking down the application, for instance, we want to add another car that stops or start differently, we can easily create a car class say Toyota that implements class Car, override it stop and start method and implement it the way we want our Toyota to start and stop. There is no need to go into the Car class to change our code whenever there is a new requirement.
+In the main method. The Car represents the interface class. Benz represents the concrete class or child class. The child class gives its memory to the Car object to have access to its implemented methods. The Benz class implemented the start and stop behaviors of the Car class in the manner it wanted. This makes our application robust. We can change or add requirements to our application without breaking down the application.
```

Comment:
> In the main method. The Car represents the interface class.

These should be a single sentence. Putting them into two sentences makes them appear to be two separate ideas when they are a part of the same phrase. Also, I have minor grammatical and formatting fixes I would like you to implement. Doing this should fix it:

> In the main method, `Car` represents the interface class.

Notice how I surround **Car** in backticks to signify it is part of the code.

Reply:
Surround any instances of the words **Car** and **Benz** in this paragraph in backticks to signify they are the names of classes in the code.